File: frontend/src/app/api/workflows.ts
import { apiGet, apiPost, qs } from "./client";

export interface WorkflowValidationSummary {
  status: "safe" | "advisory" | "invalid";
  error_count: number;
  warning_count: number;
  errors: string[];
  warnings: string[];
}

export interface WorkflowSpec {
  id: string;
  workspace_id: string;
  tenant_id: string;
  name: string;
  version: number;
  status: "draft" | "published" | "archived";
  spec: Record<string, unknown>;
  validation_summary: WorkflowValidationSummary;
  created_at: string | null;
  updated_at: string | null;
}

export interface WorkflowVersionEntry {
  id: string;
  workflow_id: string;
  version: number;
  spec: Record<string, unknown>;
  changelog: string | null;
  status: string;
  created_at: string | null;
  published_at: string | null;
  created_by: string | null;
}

export const getWorkflows = (params: {
  workspace_id: string;
  name?: string;
  status?: string;
}) => apiGet<{ items: WorkflowSpec[] }>(`/v1/workflows${qs(params)}`);

export const getWorkflow = (id: string, workspace_id: string) =>
  apiGet<WorkflowSpec>(`/v1/workflows/${id}${qs({ workspace_id })}`);

export const createWorkflow = (body: {
  workspace_id: string;
  name: string;
  spec: Record<string, unknown>;
  publish?: boolean;
}) => apiPost<WorkflowSpec>("/v1/workflows", body);

export const publishWorkflow = (id: string, workspace_id: string) =>
  apiPost<{ id: string; name: string; version: number; status: string }>(
    `/v1/workflows/${id}/publish${qs({ workspace_id })}`,
  );

export const archiveWorkflow = (id: string, workspace_id: string) =>
  apiPost<{ id: string; name: string; version: number; status: string }>(
    `/v1/workflows/${id}/archive${qs({ workspace_id })}`,
  );

export const getWorkflowVersions = (id: string, workspace_id: string, limit = 10) =>
  apiGet<{ versions: WorkflowVersionEntry[] }>(
    `/v1/versioning/workflows/${id}/versions${qs({ workspace_id, limit })}`,
  );

File: apps/platform-api-app/platform_api/schemas/workflows.py
from __future__ import annotations

from enum import Enum

from pydantic import BaseModel, Field


class WorkflowSpecStatus(str, Enum):
    draft = "draft"
    published = "published"
    archived = "archived"


class CreateWorkflowSpecRequest(BaseModel):
    workspace_id: str
    name: str = Field(min_length=2, max_length=150)
    spec: dict = Field(default_factory=dict)
    publish: bool = False


class WorkflowSpecResponseItem(BaseModel):
    id: str
    workspace_id: str
    tenant_id: str
    name: str
    version: int
    status: WorkflowSpecStatus
    spec: dict
    validation_summary: dict
    created_at: str | None = None
    updated_at: str | None = None


class WorkflowSpecPublishResponse(BaseModel):
    id: str
    name: str
    version: int
    status: WorkflowSpecStatus
    validation_summary: dict | None = None


# Keep backward-compat alias
ListWorkflowSpecsResponseItem = WorkflowSpecResponseItem
